application/models/FleetModel.php

```diff
@@ -11,5 +11,46 @@ public function __construct()
     {
         parent::__construct(APPPATH . DATA_FLEET, 'id');
     }
-    
+
+    /**
+     * Returns all planes in Lighning Air's fleet
+     */
+    public function all()
+    {
+        $planes = parent::all();
+        $planeIds = array_column($planes, 'airplaneCode');
+        $wackyPlanes = json_decode(file_get_contents('https://wacky.jlparry.com/info/airplanes'));
+        $filteredPlanes = array_filter($wackyPlanes, function($p) use ($planeIds) {
+            return in_array($p->id, $planeIds);
+        });
+
+        foreach ($filteredPlanes as $wackyPlane)
+        {
+            foreach ($planes as $plane)
+            {
+                if ($plane->airplaneCode == $wackyPlane->id) 
+                {
+                    foreach (get_object_vars($wackyPlane) as $prop => $val)
+                        if (!property_exists($plane, $prop))
+                            $plane->$prop = $val;
+                }
+            }
+        }
+        return $planes;
+    }
+
+    /**
+     * Returns one plane in Lightning Air's fleet
+     */
+    public function get($id, $key2)
+    {
```

Comment:
There's a problem within this function. It's visible when going to the fleet page `/fleet` and clicking on any of the plane links:
![image](https://user-images.githubusercontent.com/16041656/32686643-bc46a932-c65d-11e7-9c8c-ae805e0c271c.png)

I think I've got a working solution here: c1101eafac82795f7e91d3b825ae3461663664a5